custom_model_runner/setup.py

```diff
@@ -71,5 +67,5 @@
     scripts=["bin/drum"],
     install_requires=requirements,
     extras_require=extras_require,
-    python_requires=">=3.4,<3.12",
+    python_requires=">=3.8,<3.12",
```

Comment:
Why do we have the strict max? Do we know for a fact that DRUM doesn't work with 3.12?

Reply:
If I remember correct, at the time it was added, we did not have strict versions pinning for some dependencies (maybe we still do). So in this case, some dependency can pull the latest version and break things.

Reply:
I think I've seen Poetry (incorrectly) complain about no upper bound pin on a Python Version (but if that's really true, I would suggest something like this)
```suggestion
    python_requires=">=3.8,<4.0",
```

Because PY 3.12 has been out now since end of 2023 so I don't think we should restrict folks from using it unless if we know for a fact DRUM doesn't support it.

Reply:
so, @yakov-g , do you mean that because we don't pin versions of other libraries, that when someone uses 3.12, it can break things?  do we have tests for this anywhere?

Reply:
For example we don't pin `requests` (put here any name). So if there is a requests version for 3.12, now it is not installed as DRUM doesn't support 3.12.
Once DRUM starts supporting 3.12 that `requests` may break things. 
Does it make sense? I don't remember the exact reason, why we limit Py<3.12, but likely the one i mentioned.

We also pin pandas pandas>=1.5.0,<=2.0.3, while 3.12 support starts with pandas==2.2.0...

And, "no" we don't test for different Py versions.

So need to resolve this... with `poetry` ;)

Reply:
I'm not following this logic 🤔 . The version of the `requests` package (or any package for that matter) that is installed along with `datarobot-drum` shouldn't matter with what our max `python_version` pin is.  The value of the python version pin should simply reflect the usage of Python stdlib functions and other syntax that requires or disallows certain versions of Python.

All the other dependencies or transitive deps that need to be specified when actually building a base execution env should not be the responsibility of the DRUM library to specify (it would actually be incorrect to have the drum package artificially restrict the version of say the `requests` package when it works with all known versions of the library). It should be the responsibility of building the envs to specific the correct mix of libraries and versions.

Reply:
I agree with you, and this was my logic, that DRUM shouldn't pin the version as it usually uses very basic functionality. 

But failures already happened, for example with flask and werkzeug, just during some general import when a new version of those unpinned packages was released.

So probably this not about the upper Py version, but only about pinning deps.

Anyhow, need to add tests per py version. Probably in harness it could be a pipeline with several stages, per Py version,  that runs unit tests and a couple of integration/funcional tests. I mean DRUM integration/functional, not E2E with DR.